starter-code/pizza.js

```diff
@@ -43,30 +43,119 @@ function renderPepperonni() {
+function renderGreenPeppers() {
+  document.querySelectorAll('.green-pepper').forEach(function($greenPep) {
+    if (state.greenPeppers) {
+      $greenPep.style.visibility = "visible"
+    } else {
+      $greenPep.style.visibility = "hidden"
+    }
+  })
 }
 
+// Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
 function renderWhiteSauce() {
-  // Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
+  let sauceClass = document.querySelector('.sauce')
+  if (state.whiteSauce) {
+    sauceClass.classList.add('sauce-white')
+  } else {
+    sauceClass.classList.remove('sauce-white')
+  }
 }
 
+// Iteration 2: add/remove the class "crust-gluten-free" of `<section class="crust">`
 function renderGlutenFreeCrust() {
-  // Iteration 2: add/remove the class "crust-gluten-free" of `<section class="crust">`
+  let crustClass = document.querySelector('.crust')
+  if (state.glutenFreeCrust) {
+    crustClass.classList.add('crust-gluten-free')
+  } else {
+    crustClass.classList.remove('crust-gluten-free')
+  }
 }
 
-function renderButtons() {
   // Iteration 3: add/remove the class "active" of each `<button class="btn">`
+  function renderButtons() {    
+    // Lo que mi logica me dice que haga es guardar todos los botones con
+    // sus respectivas clases en una variable. Despues, checar si contienen
+    // la clase active, si la contienen, se la quito y despues se la agrego.
+    // Esto no funciona pero no me dio tiempo para más.
+    // Inclusive me truena el programa. 
+    let pBtn = document.querySelectorAll('.btn.btn-pepperoni')
+    let mushBtn = document.querySelectorAll('.btn.btn-mushrooms')
+    let gpBtn = document.querySelectorAll('.btn.btn-green-peppers.active')
+    let sauceBtn = document.querySelectorAll('.btn.btn-sauce.active')
+    let crustBtn = document.querySelectorAll('.btn.btn-crust.active')
+    
+
+    if (pBtn.contains('active')) {
+      pBtn.classList.remove('active')
+    } else {
+      pBtn.classList.add('active')
+    }
+
+    if (mushBtn.contains('active')) {
+      mushBtn.classList.remove('active')
+    } else {
+      mushBtn.classList.add('active')
+    }
+
+    if (gpBtn.contains('active')) {
+      gpBtn.classList.remove('active')
+    } else {
+      gpBtn.classList.add('active')
+    }
+
+    if (sauceBtn.contains('active')) {
+      sauceBtn.classList.remove('active')
+    } else {
+      sauceBtn.classList.add('active')
+    }
+    
+    if (crustBtn.contains('active')) {
+      crustBtn.classList.remove('active')
+    } else {
+      crustBtn.classList.add('active')
+    } 
+    
 }
 
-function renderPrice() {
   // Iteration 4: change the HTML of `<aside class="panel price">`
-}
+function renderPrice() {
+  let list = document.querySelectorAll('li')
+  let pepButton = document.querySelector('.btn.btn-pepperonni.active')
+  
+  /*Aqui lo que quería hacer es identificar cada elemento li por
```

Comment:
Tu idea iba en buen camino, esta bien que hayas hecho if else pero no tenias que hacerlo tan profundo, con hacer una condicional que cheque si el estado es Active, seleccionar con un query selector el precio del panel y agregarle un inner text para agregar o quitar el precio, en caso contrario dejarlo como not selected.